sdcm/provision/gce/provisioner.py

```diff
@@ -0,0 +1,140 @@
+            gce_services_project_name=gce_project_name,
+            location_name=self._location,
+            data_disks=definition.data_disks
+        )
+
+        metadata_args = MetadataArgs(
+            name=definition.name,
+            username=self.cluster_params.get("gce_image_username"),
+            public_key=definition.ssh_key,
+            tags=definition.tags,
+            node_index=definition.instance_index,
+            startup_script=TestConfig().get_startup_script(),
+            cluster_name=definition.name,
+            raid_level=self.cluster_params.get("raid_level")
+        )
+
+        new_node: Node = self._vm_provider.get_or_create_instance(disk_struct_args=disk_struct_args,
+                                                                  metadata_args=metadata_args,
+                                                                  instance_definition=definition,
+                                                                  pricing_model=pricing_model,
+                                                                  params=self.cluster_params)
+
+        vm_instance = VmInstance(name=new_node.name,
+                                 region=self._location,
+                                 user_name=metadata_args.username,
+                                 public_ip_address=new_node.public_ips[0],
+                                 private_ip_address=new_node.private_ips[0],
+                                 ssh_key_name=KeyStore().get_gce_ssh_key_pair().name,
+                                 tags=definition.tags,
+                                 pricing_model=pricing_model.value,
+                                 image=definition.image_id,
+                                 _provisioner=self,
+                                 _node=new_node,
+                                 creation_time=datetime.datetime.utcnow())
+        return vm_instance
+
+    def get_or_create_instances(self, definitions: List[InstanceDefinition],
+                                pricing_model: PricingModel = PricingModel.SPOT) -> List[VmInstance]:
+        vm_instances = []
+
+        for definition in definitions:
+            vm_instances.append(self.get_or_create_instance(definition=definition, pricing_model=pricing_model))
+        return vm_instances
+
+    def terminate_instance(self, name: str, wait: bool = False) -> bool:
+        return self._vm_provider.destroy_instance(name)
+
+    def reboot_instance(self, name: str, wait: bool) -> None:
+        self._vm_provider.reboot_instance(name)
+
+    def list_instances(self) -> List[VmInstance]:
+        return self._vm_provider.list_running_instances()
+
+    def cleanup(self, wait: bool = False) -> None:
+        pass
+
+    def add_instance_tags(self, name: str, tags: Dict[str, str]) -> None:
+        self._vm_provider.add_tags(vm_instance_name=name, tags=tags)
+
+    @staticmethod
+    def list_regions():
```

Comment:
`Provisioner` API does not have such method

Reply:
I think I added this one to be complementary to `_get_available_zones()`. Either way, we don't use it, so I removed it.